File: backend/src/main/java/com/csci/tax_docs_portal/mapper/MessageMapper.java
package com.csci.tax_docs_portal.mapper;

import com.csci.tax_docs_portal.entity.Message;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;
import org.springframework.jdbc.support.rowset.SqlRowSet;

public class MessageMapper {

  public Message mapRowSetToMessage(SqlRowSet rowSet) {

    if (rowSet.next()) {
      return this.mapRowSetEntryToMessage(rowSet);
    }
    return null;
  }

  public List<Message> mapRowSetToMessages(SqlRowSet rowSet) {
    List<Message> messages = new ArrayList<>();

    while (rowSet.next()) {
      messages.add(this.mapRowSetEntryToMessage(rowSet));
    }

    return messages;
  }

  public Message mapRowSetEntryToMessage(SqlRowSet rowSet) {
    return Message.builder()
        .id(UUID.fromString(rowSet.getString("id")))
        .clientId(UUID.fromString(rowSet.getString("client_id")))
        .accountantId(UUID.fromString(rowSet.getString("accountant_id")))
        .senderType(rowSet.getString("sender_type"))
        .messageText(rowSet.getString("message_text"))
        .createdAt(
            rowSet.getTimestamp("created_at") != null
                ? rowSet.getTimestamp("created_at")
                    .toLocalDateTime()
                : null
        )
        .build();
  }
}

File: frontend/src/mapper/message-mapper.ts
import { Message } from '../models/message.ts';
import { MessageInput } from '../models/message-input.ts';

export class MessageMapper {

public mapModelToInput(model: Message): MessageInput {
    return {
      id: model.id,
      clientId: model.clientId,
      accountantId: model.accountantId,
      sender_type: model.sender_type,
      message_text: model.message_text,
      created_at: model.created_at
    };
}

public mapInputToModel(input: MessageInput): Message {
    return {
      id: input.id ?? '',
      clientId: input.clientId ?? '',
      accountantId: input.accountantId ?? '',
      sender_type: input.sender_type ?? "Client",
      message_text: input.message_text ?? '',
      created_at: input.created_at ?? ''
    };
  } 
}